src/main/java/com/epam/ta/reportportal/database/dao/LaunchRepositoryCustomImpl.java

```
@@ -282,4 +306,62 @@ private boolean hasItems(Query query) {
 		Query query = query(where(PROJECT_ID_REFERENCE).is(projectRef)).addCriteria(where(issueField).exists(true));
 		return mongoTemplate.find(query, Launch.class);
 	}
+
+    @Override
+    public Page<Launch> findLatestLaunches(String project, Queryable filter, Pageable pageable) {
+        Long total = countLatestLaunches(project, filter);
+        List<Launch> launches = findLatest(project, filter, pageable);
+        return new PageImpl<>(launches, pageable, total);
+    }
+
+    private Long countLatestLaunches(String project, Queryable filter) {
+        Long total = 0L;
+        final String countKey = "count";
+        List<AggregationOperation> operations = latestLaunchesAggregationOperationsList(project, filter);
+        operations.add(count().as(countKey));
+        Map result = mongoTemplate.aggregate(newAggregation(operations), Launch.class, Map.class)
+                .getUniqueMappedResult();
+        if (null != result && result.containsKey(countKey)) {
```

seems like we need to throw some error here instead of just returning 0